### ISSUE TITLE
Cannot System.resolve with importmap prior to System.import

### ISSUE THREAD

**Issue by k-j-kim** [systemjs/systemjs#2126]
Because import-map is only loaded on `prepareImport`, it is not possible to use `System.resolve` with import-map before `System.import`. Is this the intended behaviour?

import map load:
https://github.com/systemjs/systemjs/blob/dc9a1dad19d29c1252b322d6d15c543901dd1dbe/src/features/import-map.js#L26-L40

the only place where `prepareImport` is called:
https://github.com/systemjs/systemjs/blob/dc9a1dad19d29c1252b322d6d15c543901dd1dbe/src/system-core.js#L32-L42

perhaps it would be better to lazy load the import map on demand, but I'm guessing `prepareImport` being an async load makes it awkward?

**Comment by joeldenning** [comment]
There's a bit of history here - System.resolve has changed back and forth between synchronous and asynchronous a few times. We're trying to match whatever the browser specs decide on for `import.meta.resolve()`, which has been a bit unclear and changing (see related https://github.com/nodejs/modules/issues/458). For now, System.resolve within SystemJS@6 is synchronous.

Looking at System.resolve from a "polyfill-like" perspective, its closest equivalent is `import.meta.resolve()`, which **may only be called within a module**. In order to call it within a module, you must first load that module, which in the case of SystemJS would trigger prepareImport().

In other words, I don't think the SystemJS behavior here is violating any specs that exist. The process of resolving the import map is inherently asynchronous due to the possibility of external import maps (`<script type="importmap" src="importmap.json">`).

Regarding your specific code, here's a way you can call System.resolve without having first called System.import:

```js
System.prepareImport().then(() => {
  System.resolve('thing')
})
```

I'm closing this since I view this as expected behavior / not a bug. Feel free to comment further or reopen, though.

**Comment by k-j-kim** [comment]
Appreciate the detailed explanation @joeldenning. Makes sense when treating systemjs as a polyfill.
